share/ruby-install/package_manager.sh

```diff
@@ -32,30 +33,29 @@ function set_package_manager()
 function install_packages()
 {
 	case "$package_manager" in
-		apt)	$sudo apt-get install -y "$@" || return $? ;;
-		dnf|yum)$sudo $package_manager install -y "$@" || return $?     ;;
-		port)   $sudo port install "$@" || return $?       ;;
-		pkg)	$sudo pkg install -y "$@" || return $?     ;;
-		brew)
-			local brew_owner="$(/usr/bin/stat -f %Su "$(command -v brew)")"
-			if [ "$(whoami)" -eq "$brew_owner" ]; then
-				brew install "$@" ||
-				brew upgrade "$@" || return $?
-			else
-				sudo -u "$brew_owner" brew install "$@" ||
-				sudo -u "$brew_owner" brew upgrade "$@" || return $?
-			fi
-			;;
 		pacman)
 			local missing_pkgs=($(pacman -T "$@"))
 
 			if (( ${#missing_pkgs[@]} > 0 )); then
-				$sudo pacman -S "${missing_pkgs[@]}" || return $?
+				installer -S "${missing_pkgs[@]}" || return $?
 			fi
 			;;
-		zypper) $sudo zypper -n in -l $* || return $? ;;
-		"")	warn "Could not determine Package Manager. Proceeding anyway." ;;
+		"")	    warn "Could not determine Package Manager. Proceeding anyway." ;;
+		brew)   installer install "$@" || installer upgrade "$@" || return $? ;;
+		zypper) installer -n in -l $* || return $? ;;
+		*)      installer install -y "$@" || return $?;;
 	esac
 }
 
+function installer() {
+	p="$(command -v $package_manager)"
```

Comment:
You need the `local` keyword to prevent leaking variables.

Reply:
Good catch.